miniwallet/contracts/proxy/ERC1967Proxy.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: Apache-2.0
+
+// OpenZeppelin Contracts v4.4.1 (proxy/ERC1967/ERC1967Proxy.sol)
+
+pragma solidity ^0.8.0;
+
+import "hardhat-deploy/solc_0.8/openzeppelin/proxy/Proxy.sol";
+import "hardhat-deploy/solc_0.8/openzeppelin/proxy/ERC1967/ERC1967Upgrade.sol";
+
+/**
+ * @dev This contract implements an upgradeable proxy. It is upgradeable because calls are delegated to an
+ * implementation address that can be changed. This address is stored in storage in the location specified by
+ * https://eips.ethereum.org/EIPS/eip-1967[EIP1967], so that it doesn't conflict with the storage layout of the
+ * implementation behind the proxy.
+ */
+contract ERC1967Proxy is Proxy, ERC1967Upgrade {
```

Comment:
As per discussion with @polymorpher and @johnwhitton on Sep 12th.
We should expose additional functions such as `_authorizeUpgrade` and interact directly with the proxy contract to do the upgrades rather than using hardhat-deploy. 
This will give us greater control of the deployment process